app/scripts/store.js

```diff
@@ -97,9 +97,6 @@ var exports = (function() {
             resolve(json);
           })
           .fail(function(error) {
-            if (error.status === 404) {
-              resolve(undefined);  // expected behaviour as too strict filtering results in a 404 (as you're filtering out the model you're requesting): resolve with undefined
-            }
```

Comment:
I removed this, as we do not do anything with `undefined` or `error`. Preparation for a proper handling of Promise fails.